src/aioswitcher/api/helper.py

```diff
@@ -0,0 +1,46 @@
+# Copyright Tomer Figenblat.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Switcher integration helper functions."""
+
+from asyncio import StreamReader, StreamWriter
+from binascii import unhexlify
+from logging import getLogger
+
+from ..device.tools import set_message_length, sign_packet_with_crc_key
+
+logger = getLogger(__name__)
+
+
+async def send_packet(
+    writer: StreamWriter, reader: StreamReader, packet_id: str, packet: str
+) -> bytes:
+    """Sign and send a packet, then read the response.
```

Comment:
It would be better to add this as a private class method (`_send_packet`) under the `SwitcherApi`
- It is not used outside of this class
- As a member of the class it will have access to the `writer` & `reader` streams so you don't have to pass them

Reply:
You are probably right, but it will miss the point when we want to decrease that file length.
If so, I must move this from helper to the same file Where SwitcherApi is.

Reply:
With this PR you gather few methods which are used very often in a single method, this should reduce the size.
When we have all the common parts optimized in methods we can later try to split the transport into another class if it will make sense